src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
+
+  if (fromFormatObject.yy !== toFormatObject.yy) {
+    if (fromFormatObject.yy.length === 4) {
+      dateObject.yy = dateObject.yy.slice(2);
+    } else {
+      if (+dateObject.yy < 30) {
+        dateObject.yy = '20' + dateObject.yy;
+      } else {
+        dateObject.yy = '19' + dateObject.yy;
+      }
+    }
+  }
+
+  return resultDate(dateObject, toFormat);
+}
+
+function resultDate(dateObject, toFormat) {
+  const result = [];
+
+  for (let i = 0; i < 4; i++) {
```

Comment:
The loop in the `resultDate` function is hardcoded to iterate 4 times, which assumes that the format arrays always have 4 elements. This might not be flexible if the format arrays have a different structure. Consider iterating based on the length of the `toFormat` array.